src/Type/ObjectType.php

```diff
@@ -1384,7 +1386,7 @@ public function changeSubtractedType(?Type $subtractedType): Type
 				$subtractedTypesList = TypeUtils::flattenTypes($subtractedType);
 				$subtractedTypes = [];
 				foreach ($subtractedTypesList as $type) {
-					$subtractedTypes[$type->describe(VerbosityLevel::precise())] = $type;
```

Comment:
I think we don't need the key of this array and building up the descriptions shows up in profiles of phpstan/phpstan#11263

Reply:
after getting rid of the key, it seems we can get rid of the whole loop